java-rmi/src/test/java/com/baeldung/rmi/JavaRMIIntegrationTest.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.rmi;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.rmi.NotBoundException;
+import java.rmi.RemoteException;
+import java.rmi.registry.LocateRegistry;
+import java.rmi.registry.Registry;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class JavaRMIIntegrationTest {
+	
+	@BeforeClass
+	public static void whenRunServer_thenServerStarts() {
```

Comment:
Sorry, this was unclear, but doesn't this method name seem weird now that it isn't a test method? I guess I'd expect a name like "serverStartup" or "setUp". It also seems a little odd that we are asserting something inside a setup method.

Reply:
Doesn't satisfy the naming rules.

Reply:
I appreciate your careful reading of the naming rules, but that rule only applies to tests. By annotating this with @BeforeClass, it is no longer a test, but is instead a setup method. If you want to change it back to a @Test, then that is one way to solve the issue as well; you simply need to configure each test so that it is executed in the correct order.

Reply:
There is no such mention in the guidelines. So, If you could please confirm it, I will do the change. Just don't want to do same changes again.